feat: show 2FA secret key for manual entry during setup

zlwaterfield opened the pull request:
> [!IMPORTANT]
> 👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changes

We currently only render the 2FA QR code, so if a user is using a password manager that doesn't support QR codes, they can't set it up. This returns the secret to the client, allowing them to use it as a backup if needed. 

<img width="546" height="653" alt="Screenshot 2025-07-11 at 1 16 11 PM" src="https://github.com/user-attachments/assets/c27ae2f5-3a64-4d5e-8288-a2cb28093e3e" />


## Did you write or update any docs for this change?

<!-- Engineers are responsible for doing the first pass at documenting their features and/or code.  -->

- [ ] I've [added or updated the docs](https://posthog.com/handbook/engineering/writing-docs)
- [ ] I've reached out for help from the docs team
- [ ] No docs needed for this change

## How did you test this code?

Manually

zlwaterfield committed:
Show 2FA secret key for manual entry during setup

The backend now returns the 2FA secret key when starting setup, and the frontend displays it for users who cannot scan the QR code. This improves accessibility for users setting up two-factor authentication.

zlwaterfield committed:
Update twoFactorLogic.ts

Piccirello reviewed (approved):
This is a nice improvement!

Btw the current TOTP secret is 20 hex characters, or 80 bits of entropy. 128 bits minimum is recommended, and 160 bits would be more ideal. Of course the downside with hex encoding is that 160 bits of entropy equates to a 40 character string. Bumping the length to at least 32 to satisfy the 128 bit minimum would be great, though of course non blocking for this PR.

zlwaterfield committed:
Update two factor setup test to expect secret in response

The test for two factor start setup now checks that the response includes both 'success' and a 'secret' key, reflecting updated API behavior.

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)
